relay-event-schema/src/protocol/span.rs

```diff
@@ -201,11 +201,11 @@ pub struct SpanData {
 
     /// The sentry environment.
     #[metastructure(field = "environment")]
-    pub environment: Annotated<Value>,
+    pub environment: Annotated<String>,
 
     /// The release version of the project.
     #[metastructure(field = "release")]
-    pub release: Annotated<Value>,
+    pub release: Annotated<LenientString>,
```

Comment:
I typed out these fields so they could be more easily converted to and from their transaction counterpart. They were previously `Value` because we did not want to fail span deserialization of the entire span if the field contains something else. `span.data` used to be an untyped data bag, but is becoming more and more narrowly defined as we adopt Otel conventions (see https://develop.sentry.dev/sdk/performance/span-data-conventions/).